### ISSUE TITLE
Friendly modules names when hot reloading?

### ISSUE THREAD

**Issue by glenjamin** [webpack/webpack#1129]
I tried to search the docs for this, but couldn't find anything.

Is there any way to ask webpack to retain filenames, so that instead of:

```
[HMR] Checking for updates on the server...
[HMR] Updated modules:
[HMR]  - 240
[HMR]  - 119
```

These numbers display as the filenames?

I'm happy to write up a proper docs PR if someone can point me in the right direction.

**Comment by karlnp** [comment]
https://github.com/webpack/webpack/tree/master/examples/labeled-modules

**Comment by glenjamin** [comment]
Labelled modules appears to be a different module format, rather than a way to get existing modules names in HMR log output.

**Comment by sokra** [comment]
Currently there is not way to get the names there. But you can look them up in your stats.

**Comment by glenjamin** [comment]
Have managed to incorporate this into webpack-hot-middleware v2 - https://github.com/glenjamin/webpack-hot-middleware#200

Had to replace `dev-server` and `log-apply-updates` with my own equivalents, but seems to work pretty well

**Comment by glenjamin** [comment]
Is this being wontfixed? I still think it's a useful improvement, and would be far more efficient in webpack itself.

**Comment by bebraw** [comment]
Alright. I labeled this accordingly.

**Comment by bebraw** [comment]
By the way, enabling `new webpack.NamedModulesPlugin()` for development can be a good idea. That gives me `[HMR]  - ./app/component.js` kind of updates and makes it easier to debug the dev code anyhow.

**Comment by glenjamin** [comment]
`NamedModulesPlugin` sounds perfect - is this a fairly new addition? I don't think it existed when this issue was first opened.

I'll give it a go - sounds like it does exactly what this issue requested.

**Comment by bebraw** [comment]
Webpack 2, yup.

**Comment by bebraw** [comment]
Never mind. It's in webpack 1 too.

**Comment by glenjamin** [comment]
Looks like this got added in https://github.com/webpack/webpack/commit/a04ffb928365b19feb75087c63f13cadfc08e1eb which was first released in `v1.10.0` (approx 3 weeks after https://github.com/webpack/webpack/issues/1129#issuecomment-109789979)

I think this can safely be called closed fixed then.

Is there a tracking issue for documenting the existence / reason / usage of core plugins like this?

**Comment by bebraw** [comment]
Not really. Feel free to open issues at the webpack.js.org repo, though.

`NamedModulesPlugin` would be great development default even. Maybe something for webpack 3.
